portfolio/src/main/webapp/script.js

```diff
@@ -32,17 +32,30 @@ function addRandomFact() {
  * Fetch the list of comments and add them to the page
  */
 function fetchAndAddComments(){
-  fetch('/comments').then(response => response.json()).then((comments) => {
+  fetch('/comments').then(response => response.json()).then((comment) => {
     const commentContainer = document.getElementById('comments-container');
-    comments.forEach(function(comment) {
-      commentContainer.appendChild(createListElement(comment))
+    comment.forEach(function(comment) {
+      commentContainer.appendChild(createCommentElement(comment))
     });
+
   });
 }
 
-/** Creates and returns an <li> element containing text. */
-function createListElement(text) {
-  const liElement = document.createElement('li');
-  liElement.innerText = text;
-  return liElement;
-}
\ No newline at end of file
+/** Creates and returns a comment element containing  the components
+ of a comment. */
+function createCommentElement(comment) {  
+  const commentUsername = document.createElement('username');  
```

Comment:
I dont think you're using createElement correct and javascript is simply producing undefined behavior because it doesnt know how to fail. createElement takes as a parameter the html element you want to create. There is no element called "username" or date". I think you mean:
document.createElement("div");

See https://www.w3schools.com/jsref/met_document_createelement.asp, ignore the upper case that they use